### ISSUE TITLE
Size of arguments calling poly

### ISSUE THREAD

**Issue by edljk** [JuliaPlots/Makie.jl#1514]
I am a little bit confused with the size of attribute using a vector of polygons:

```julia
using GLMakie
P = [Polygon([Point2(rand(), rand()) for _ = 1:3]) for k = 1:2]
poly!(P, color = [:red, :green], strokecolor = :blue, strokewidth = 1)
```
works fine. Surprisingly, choosing one different color by triangle

```julia
poly!(P, color = [:red, :green], strokecolor = [:blue, :red], strokewidth = 1)
```
produces an error. Is it the expected behavior?
Thanks!

**Comment by SimonDanisch** [comment]
Hm, guess strokecolor isn't handled correctly by the multi poly recipe...

**Comment by edljk** [comment]
Not sure it is a related issue.
Since update 0.16.1, the following instructions produce a strange error

```julia
using Makie, GLMakie

y, z, sizes = rand(10), rand(10), 30 * rand(10)
f = Figure()
Axis(f[1, 1])
scatter!(y, z,  markersize = sizes, marker = :circle)

ERROR: UndefVarError: x not defined
Stacktrace:
  [1] (::GLMakie.GLVisualize.var"#51#52"{Vec{2, Float32}})(xs::Vec{2, Float32})
    @ GLMakie.GLVisualize $HOME/.julia/packages/GLMakie/oDJ8D/src/GLVisualize/visualize/particles.jl:292
  [2] iterate
    @ ./generator.jl:47 [inlined]
  [3] _collect(c::Vector{Vec{2, Float32}}, itr::Base.Generator{Vector{Vec{2, Float32}}, GLMakie.GLVisualize.var"#51#52"{Vec{2, Float32}}}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base ./array.jl:744
  [4] collect_similar
...
```
Sorry not to be able to investigate further, thanks.

**Comment by edljk** [comment]
It seems that this line

[particles.jl](https://github.com/JuliaPlots/Makie.jl/blob/413be2c1aaecafd4ba5bb11c7b8952e8a67e7b07/GLMakie/src/GLVisualize/visualize/particles.jl#L292)

should be replaced by

```julia 
map(x -> f * x, xs)
```

**Comment by edljk** [comment]
Btw, the original problem does not seem to be resolved in Makie 0.16.1

```julia
using GeometryBasics, GLMakie
f = Figure(); Axis(f[1, 1])
P = [Polygon([Point2(rand(), rand()) for _ = 1:3]) for k = 1:2]
poly!(P, color = [:red, :green], strokecolor = [:blue, :red], strokewidth = 1)

ERROR: Buffer vertex does not have the same length as the other buffers.
	color has length 2
	vertex has length 9
	valid_vertex has length 9
...
```

**Comment by ffreyer** [comment]
> It seems that this line
> 
> [particles.jl](https://github.com/JuliaPlots/Makie.jl/blob/413be2c1aaecafd4ba5bb11c7b8952e8a67e7b07/GLMakie/src/GLVisualize/visualize/particles.jl#L292)
> 
> should be replaced by
> 
> ```julia
> map(x -> f * x, xs)
> ```

Oh good catch. Do you want to make a pr for this?
